src/p2p/net_node.inl

```diff
@@ -1226,19 +1226,53 @@ namespace nodetool
       size_t random_index;
       const uint32_t next_needed_pruning_stripe = m_payload_handler.get_next_needed_pruning_stripe().second;
 
+      // build a set of all the /16 we're connected to, and prefer a peer that's not in that set
+      std::set<uint32_t> classB;
+      if (&zone == &m_network_zones.at(epee::net_utils::zone::public_)) // at returns reference, not copy
+      {
+        zone.m_net_server.get_config_object().foreach_connection([&](const p2p_connection_context& cntxt)
+        {
+          if (cntxt.m_remote_address.get_type_id() == epee::net_utils::ipv4_network_address::get_type_id())
+          {
+
+            const epee::net_utils::network_address na = cntxt.m_remote_address;
+            const uint32_t actual_ip = na.as<const epee::net_utils::ipv4_network_address>().ip();
```

Comment:
Does epee always return IP addresses in little-endian byte order, or is this machine-specific?

Reply:
It appears so. See https://github.com/monero-project/monero/pull/5544/commits/04a103be2d5174b3febadd296215a9cf2b3a10dc, and I'm noticing that the comment there is wrong, they're kept little endian, not network byte order. It was confusing to get it right...

Reply:
That commit doesn't give me much confidence - it looks like addresses are kept in machine-specific byte order, and the rest of the code then treats it as little-endian, meaning that there are more cases to fix on big-endian machines.

Reply:
I would totally accept a patch which straightens all this out :)

Still, monerod syncs fine with the patches in the PR the link above is from, so it's progress.